site/src/sphinx/client-service-discovery.rst

```diff
@@ -0,0 +1,237 @@
+.. _client-service-discovery:
+
+Client-side load balancing and service discovery
+================================================
+
+You can configure an Armeria client to distribute its requests to more than one server autonomously, unlike
+traditional server-side load balancing where the requests goes through a dedicated load balancer such as
+`L4 and L7 switches <https://en.wikipedia.org/wiki/Multilayer_switch#Layer_4%E2%80%937_switch,_web_switch,_or_content_switch>`_.
+
+There are 4 elements involved in client-side load balancing in Armeria:
+
+- :api:`Endpoint` represents an individual host (with an optional port number) and its weight.
+- :api:`EndpointGroup` represents a set of :apiplural:`Endpoint`.
+- :api:`EndpointGroupRegistry` is a global registry of :apiplural:`EndpointGroup` where each
+  :api:`EndpointGroup` is identified by its unique name.
+- A user specifies the target group name in a URI,
+  e.g. ``http://group:my_group/`` where ``my_group`` is the group name.
+
+.. uml::
+
+    @startuml
+    hide empty members
+
+    class EndpointGroupRegistry <<singleton>> {
+        groups : Map<String, EndpointGroup>
+    }
+
+    class EndpointGroup {
+        endpoints : Set<Endpoint>
+    }
+
+    class Endpoint {
+        host : String
+        port : int
+        weight : int
+    }
+
+    EndpointGroupRegistry o-right- "*" EndpointGroup
+    EndpointGroup o-right- "*" Endpoint
+    @enduml
+
+.. _creating-endpoint-group:
+
+
+Creating an ``EndpointGroup``
+-----------------------------
+There are various :api:`EndpointGroup` implementations provided out of the box, but let's start simple with
```

Comment:
let's simply start with?

Reply:
I guess 'start simple' is OK here. e.g. think big, act small